### PR TITLE
Add slim coverage check running on local (branch) copy CL

### LINKED ISSUE #2177: Add slim coverage check as GitHub Action running on local (branch) copy CL
Running against UberGraph means that changes to the ontology are not reflected in the report.  Could we acheieve this by running RelationGraph for the relevant relation types in a branch and then running SPARQL queries as we currently do against UberGraph.

CC @bvarner-ebi

### PR DESCRIPTION
**Pull request by ubyndr**
Resolves #2177 

We decided not to follow the GitHub action route in this issue. Editors should use the following commands to run coverage reports via the Makefile in the ontology directory.;

- for specific upper_slim report:
```sh
sh run.sh make reports/general_cell_types_upper_slim_report.csv
```
```sh
sh run.sh make reports/blood_and_immune_upper_slim_report.csv
```
```sh
sh run.sh make reports/eye_upper_slim_report.csv
```
- for all upper_slim reports:
```sh
sh run.sh make slim_coverage
```

cc @aleixpuigb @bvarner-ebi

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,4 @@
 mkdocs
+
+rdflib~=7.0.0
+SPARQLWrapper~=2.0.0
```
Comment (anitacaron) [inline requirements.txt:4]: This is part of the ODK, you don't need to install :)

File: src/scripts/requirements.txt
Comment (anitacaron) [inline src/scripts/requirements.txt:1]: empty requirements.txt file?

File: requirements.txt (hunk 1 above)
Comment (ubyndr) [reply requirements.txt:4]: Bradley wanted to use script itself without running it through make command, so it is kinda needed, isn't it?

Comment (anitacaron) [reply requirements.txt:4]: is there a reason?

Comment (anitacaron) [reply requirements.txt:4]: He can run the command without make but still use ODK, which is recommended.

File: src/scripts/requirements.txt
Comment (anitacaron) [reply src/scripts/requirements.txt:1]: in any case, I don't see this is needed.

File: requirements.txt (hunk 1 above)
Comment (ubyndr) [reply requirements.txt:4]: cc @bvarner-ebi

**Review by aleixpuigb — DISMISSED** [review]
The tests performed have been successful, with a running time of 20 minutes.

File: src/scripts/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+rdflib~=7.0.0
```
Comment (anitacaron) [inline src/scripts/requirements.txt:1]: really need this?

Comment (ubyndr) [reply src/scripts/requirements.txt:1]: I had deleted it from my local branch, dunno why it still exists in the PR...

File: src/scripts/generic_coverage.py [hunk 3]
```diff
@@ -98,7 +127,6 @@ def get_scope_query(scope_term: str) -> str:
         WHERE
         {{
           ?scope_member rdfs:subClassOf|BFO:0000050|RO:0002100 {_scope} .
```
[the remark this answers is not in the document]
Comment (ubyndr) [reply src/scripts/generic_coverage.py:129]: relation-graph command should help with these, shouldn't it? Did I misinterpreted the command's function?

**Review by aleixpuigb — CHANGES_REQUESTED** [review]
I have test the eye slim with `eye` UBERON:0000970 this time and the coverage includes UBERON terms which should not. Moreover, it doesn't cover many cell types that it used to. Are the object properties overlaps or 'has soma location' (subproperty of overlaps) included?
The running time was 13:30 minutes.

File: src/scripts/generic_coverage.py [hunk 4]
```diff
@@ -124,7 +152,6 @@ def get_superclass_value_query(term_iri_list: List[str], _scope: str) -> str:
             {{
               ?term rdfs:subClassOf ?super. ?super rdfs:label ?label. 
               ?super rdfs:subClassOf|BFO:0000050 {_scope}.
-              ?super rdfs:isDefinedBy <http://purl.obolibrary.org/obo/cl.owl> .
```
Comment (anitacaron) [inline src/scripts/generic_coverage.py:127]: Should change it to filter:

```
FILTER (STRSTARTS(STR(?super), "http://purl.obolibrary.org/obo/CL_"))
```

File: src/scripts/generic_coverage.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/scripts/generic_coverage.py:129]: Need to double-check.

**Review by bvarner-ebi — APPROVED** [review]
Report successfully generated

**Review by aleixpuigb — APPROVED** [review]
The report has been generated with the expected coverage and uncovered terms.

**Pull request merged**